core/lib/dal/src/transactions_dal.rs

```diff
@@ -571,7 +608,211 @@ impl TransactionsDal<'_, '_> {
         }
     }
 
-    async fn handle_executed_l2_transactions(
+    async fn insert_executed_l2_transactions(
+        &mut self,
+        l2_block_number: L2BlockNumber,
+        block_base_fee_per_gas: U256,
+        transactions: &[TransactionExecutionResult],
+    ) -> DalResult<()> {
+        let l2_txs_len = transactions
+            .iter()
+            .filter(|tx_res| {
+                matches!(
+                    tx_res.transaction.common_data,
+                    ExecuteTransactionCommon::L2(_)
```

Comment:
why do l1 transactions need special treatment? Please add a comment

Reply:
Same as with updating: l2/l1/upgrade txs have different sets of fields that need to be inserted/updated